homeassistant/components/homekit/type_fans.py

```diff
@@ -126,7 +128,7 @@ def __init__(self, *args: Any) -> None:
                 )
 
                 def setter_callback(value: int, preset_mode: str = preset_mode) -> None:
-                    return self.set_preset_mode(value, preset_mode)
+                    self.set_preset_mode(value, preset_mode)
 
```

Comment:
Looks like we need to suppress the Auto preset mode here if its an air purifier

Reply:
Ie don't add the switch for Auto if its an air purifier

Reply:
Can you elaborate on that? Looking at `AirPurifier.set_chars`, the air purifier itself also uses `set_preset_mode` when switching to an auto preset.

Reply:
<img width="276" alt="Screenshot 2025-04-08 at 1 53 30 AM" src="https://github.com/user-attachments/assets/deede1a6-7c0d-48e1-bfd5-49f9320d634e" />
<img width="499" alt="Screenshot 2025-04-08 at 1 53 37 AM" src="https://github.com/user-attachments/assets/dd1d3f08-849e-4221-a957-319b6e97e5ec" />

Reply:
Its still creating the Auto switch